software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessWinRmDriver.java

```diff
@@ -129,10 +132,40 @@ protected WinRmToolResponse executeCommand(ConfigKey<String> regularCommandKey,
             throw new IllegalStateException(String.format("%s and %s cannot both be set", regularCommandKey.getName(), powershellCommandKey.getName()));
         }
 
+        WinRmToolResponse response;
         if (Strings.isNullOrEmpty(regularCommand)) {
-            return getLocation().executePsScript(ImmutableList.of(powershellCommand));
+            response = getLocation().executePsScript(ImmutableList.of(powershellCommand));
         } else {
-            return getLocation().executeScript(ImmutableList.of(regularCommand));
+            response = getLocation().executeScript(ImmutableList.of(regularCommand));
+        }
+
+        if (Tasks.current()!=null) {
```

Comment:
I'd cache whether we've made use of these stdOut/stdErr, rather than calling `if (Tasks.current()!=null)` again. If we added those tags previously then call `writeToStream` (rather than also looking up the `BrooklynTaskTags.stream()` again - no need to do that).